promql/value.go

```diff
@@ -101,6 +107,34 @@ func (s Series) MarshalJSON() ([]byte, error) {
 	return json.Marshal(series)
 }
 
+func (s Series) writeArrow(w io.Writer, pool memory.Allocator) error {
+	fields := []arrow.Field{
+		{Name: "t", Type: &arrow.TimestampType{Unit: arrow.Millisecond}},
+		{Name: "v", Type: arrow.PrimitiveTypes.Float64},
+	}
+	metadata := arrow.MetadataFrom(s.Metric.Map())
```

Comment:
Is there a specific reason to put the metric as map into the metadata? 
I probably would have put it into the schema itself. I can see both work, to be honest.

Reply:
The reason for using Metadata is mostly how we use it internally now for transferring time series data between a few services. I guess the other option would be to add it as the name of the value column and have the client parse it as JSON?